### ISSUE TITLE
kornia.utils.draw_line not working on horizontal lines

### ISSUE THREAD

**Issue by jere357** [kornia/kornia#2188]
### Describe the bug

Hello, i have a bizarre error where the draw_line function behaves weirdly when the line parameter A==0;
i have not tested how it behaves on vertical lines

### Reproduction steps

```bash
#my input is exactly this:
line = tensor([ 180,  328, 1037,  328], dtype=torch.int32)
im_drawn of torch.Size([3, 544, 1056])
#and i call the function like this: (keep in mind it draw all lines normally but misbehaves only when y1==y2)
im_drawn = kornia.utils.draw_line(im_drawn,
        torch.tensor([line[0],line[1]]),
        torch.tensor([line[2], line[3]]),
        color=torch.tensor([255,0,0], dtype=torch.uint8))

#and so when it gets to the if A==0 in the draw_line part it breaks down with this error
RuntimeError: The expanded size of the tensor (858) must match the existing size (3) at non-singleton dimension 1.  Target sizes: [3, 858].  Tensor sizes: [3]
```

### Expected behavior

i expect it to draw the horizontal line normally.

### Environment

```shell
my enviroment is the yolov5 docker + i installed kornia on it.

Collecting environment information...
PyTorch version: 1.13.0a0+936e930
Is debug build: False
CUDA used to build PyTorch: 11.8
ROCM used to build PyTorch: N/A

OS: Ubuntu 20.04.5 LTS (x86_64)
GCC version: (Ubuntu 9.4.0-1ubuntu1~20.04.1) 9.4.0
Clang version: Could not collect
CMake version: version 3.24.1
Libc version: glibc-2.31

Python version: 3.8.10 (default, Jun 22 2022, 20:18:18)  [GCC 9.4.0] (64-bit runtime)
Python platform: Linux-4.15.0-197-generic-x86_64-with-glibc2.29
Is CUDA available: True
CUDA runtime version: 11.8.89
CUDA_MODULE_LOADING set to: LAZY
GPU models and configuration: GPU 0: NVIDIA GeForce RTX 3090
Nvidia driver version: 515.65.01
cuDNN version: Probably one of the following:
/usr/lib/x86_64-linux-gnu/libcudnn.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_infer.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_adv_train.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_infer.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_cnn_train.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_infer.so.8.7.0
/usr/lib/x86_64-linux-gnu/libcudnn_ops_train.so.8.7.0
HIP runtime version: N/A
MIOpen runtime version: N/A
Is XNNPACK available: True

Versions of relevant libraries:
[pip3] functorch==1.13.0a0+936e930
[pip3] numpy==1.22.2
[pip3] pytorch-quantization==2.1.2
[pip3] torch==1.13.0a0+936e930
[pip3] torch-tensorrt==1.3.0a0
[pip3] torchvision==0.15.0a0
[conda] Could not collect
```

### Additional context

I have "fixed" the error for my case by modifying the draw_line code like so:

after this part in your code
\# calcullate coefficients A,B,C of line
\# from equation Ax + By + C = 0
A = y2 - y1
B = x1 - x2
C = x2 * y1 - x1 * y2
#i have added this line to slightly bend my line so it could be drawn since i don't really care about it being drawn precisely.
#the formatting of this if statement is weird here on github idk
if A == 0:
    y2 = y2 + 1
    y1 = y1 - 1
    A = y2-y1
    C = x2 * y1 - x1 * y2

after this it started drawing lines in a satisfying way for me, but im not sure why it does not work when A==0 . It calls this code
image[:, y1, x1 : x2 + 1] = color
intuitively it makes sense to me but im not sure why it breaks down

**Comment by edgarriba** [comment]
This is weird since that case Is covered by test
https://github.com/kornia/kornia/blob/master/test/utils/test_draw.py#L34

Feel free to send a PR if it’s really a bug

**Comment by jere357** [comment]
hmm, it is pretty weird idk, my fix is kinda janky bcs the resulting line is not a horizontal line anymore, i bend it slightly, the only difference i can see between my case and the test case is that my img tensor has 3 channels while the test case tensor has only 1 channel that could be the difference idk why the mysterious 3 appears in the error i get.

**Comment by edgarriba** [comment]
you are right, I'd suggest to fix and debug to extend the existing tests for C > 1

**Comment by cjpurackal** [comment]
closed via #2373
